### PR TITLE
Remove `rem()` function from public sass

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Part of
* #4605

### CODE REVIEW

File: src/components/ActionList/ActionList.scss
Comment (aaronccasanova) [inline src/components/ActionList/ActionList.scss]: **thought:** Per the recent discussions around token prefixes, I think these need to be renamed to `--pc-actionlist-`.

File: src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss
Comment (aaronccasanova) [inline src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:27]: I'm not sure why we were converting `0` to `rem`s, but should we add `px` to `0` here?

File: src/components/Autocomplete/components/MappedAction/MappedAction.scss
Comment (aaronccasanova) [inline src/components/Autocomplete/components/MappedAction/MappedAction.scss]: This is now an interesting mix of `css` `calc` math and `scss` math. Should we put everything in `calc` and add parens to maintain the same order of operations?

File: src/components/Autocomplete/components/MappedAction/MappedAction.scss
Comment (aaronccasanova) [inline src/components/Autocomplete/components/MappedAction/MappedAction.scss]: Same question as above in regards to custom property prefixing ☝️

`--pc-autocomplete-`?

File: src/components/Badge/Badge.scss
Comment (aaronccasanova) [inline src/components/Badge/Badge.scss:20]: `0.1px` - Wonder if this was intentional?

File: src/components/BulkActions/BulkActions.scss
Comment (aaronccasanova) [inline src/components/BulkActions/BulkActions.scss]:
```suggestion
  transform: translateX(calc(-1 * var(--p-space-10)));
```

File: src/components/Choice/Choice.scss
Comment (aaronccasanova) [inline src/components/Choice/Choice.scss:62]: **question:** Do we no longer need the fallback?

File: src/components/ChoiceList/ChoiceList.scss
Comment (aaronccasanova) [inline src/components/ChoiceList/ChoiceList.scss:21]: Same as above. Do we need the fallback?

File: src/components/IndexTable/IndexTable.scss
Comment (aaronccasanova) [inline src/components/IndexTable/IndexTable.scss:379]: I'm not used to seeing decimals for `px` units. Similar to the above comment, do you think this is intentional?

File: src/styles/foundation/_focus-ring.scss
Comment (aaronccasanova) [inline src/styles/foundation/_focus-ring.scss]: Our regex may have unintentionally updated this comment/variable. We should probably tophat this variable, make sure it works as expected, and update or remove the comment. Thoughts?

**Review by aaronccasanova — APPROVED** [review]
Nice work Aveline! Regex FTW 🙌 

I tried to be as thorough as possible, but would recommend waiting for another set of eyes before merging 🙂

File: src/components/Icon/Icon.scss
Comment (laurkim) [inline src/components/Icon/Icon.scss:6]: Can we use `var(--p-space-5)` here?

File: src/components/IndexTable/components/Checkbox/Checkbox.scss
Comment (laurkim) [inline src/components/IndexTable/components/Checkbox/Checkbox.scss:4]: Can we use `var(--p-space-8)` here?

File: src/components/Badge/Badge.scss
Comment (aveline) [reply src/components/Badge/Badge.scss:20]: Since it's for print, I would think yes.

**Review by laurkim — APPROVED** [review]
nice! 💯 ✨ 

Only suggestion I have is to update the naming convention for custom component properties throughout the files (`--pc-[component]-`) that was mentioned in the [issue](https://github.com/Shopify/polaris-react/issues/4732#issuecomment-982859990), like @aaronccasanova called out.

File: src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss
Comment (aveline) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:27]: Because it's `0` I don't think a unit is necessary

File: src/components/Choice/Choice.scss
Comment (aveline) [reply src/components/Choice/Choice.scss:62]: I don't think so since custom properties are well supported now. @kyledurand are there any other reasons we might still need the fallback?

File: src/components/IndexTable/IndexTable.scss
Comment (aveline) [reply src/components/IndexTable/IndexTable.scss:379]: I agree decimals for `px` is unusual, but I'm inclined to leave as is for the scope of this PR. We have intentions to review hardcoded values so might make more sense to review at that point.

File: src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss
Comment (aaronccasanova) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:27]: Okay perfect, me either. Was mainly inquiring because of the previous implementation 👍

File: src/components/Badge/Badge.scss
Comment (aaronccasanova) [reply src/components/Badge/Badge.scss:20]: Ah I think you're right that `print` is the important distinction here! It's not easy finding good reading material on this topic, but managed to find this [w3.org article on css units](https://www.w3.org/Style/Examples/007/units.en.html) that backs this up.

> In fact, CSS requires that 1px must be exactly 1/96th of an inch in all printed output. CSS considers that printers, unlike screens, do not need to have different sizes for px in order to print sharp lines. In print media, a px thus not only has the same visual appearance from one device to another, but indeed it is measurably the same.

File: src/components/IndexTable/IndexTable.scss
Comment (aaronccasanova) [reply src/components/IndexTable/IndexTable.scss:379]: Oh that's perfect! Yes, let's address this later 👌

**Review by alex-page — APPROVED** [review]

**Pull request merged**
